crates/flox/src/commands/package.rs

```diff
@@ -205,6 +208,34 @@ pub(crate) mod interface {
 
     #[derive(Bpaf, Clone, Debug)]
     pub struct PublishV2 {
+        /// Prefer https access to repositories published with a `github:` reference
+        #[bpaf(long)]
+        pub prefer_https: bool,
+
+        /// Signing key file to sign the binary with
+        ///
+        /// When omitted, reads from the config
+        #[bpaf(long, short('k'))]
+        pub sign_key: Option<PathBuf>,
+
+        /// Url of a binary cache to push binaries _to_
+        ///
+        /// When omitted, reads from the config
+        #[bpaf(long, short('c'))]
+        pub cache_url: Option<SubstituterUrl>,
+
+        /// Url of a substituter to pull binaries _from_
+        ///
+        /// When ommitted, uses the value for cache-url and falls back to the config
+        #[bpaf(long, short('s'))]
+        pub substituter_url: Option<SubstituterUrl>,
```

Comment:
Optional (bikeshed warning): any reason to rename these from `--upload-to` and `--download-from`? Those feel like words anyone will understand, which is not true of substituter.

Reply:
Apart from that I find the the names a bit clunky, the similarity of the names suggests that I need to set both, at which point you still have to explain the differences.
`upload_to` and `download_from` also make weird config keys (verbs, rather than nouns) that require additional context.
For them to work in the config it would need to be `publish.upload_to` I think.

I used "cache" and "substituter" because those seem more descriptive depending on your position; As a developer you would want to cache binaries, while at as a consumer you'd want them to be provided by a substituter.
If we don't want to use the word substituter for some reason, it might be something like `providing-cache-url`?

But to cut the bikeshedding short, either way seems somewhat reasonable, I don't have strong preferences except against putting `upload_to` into the config without context